state/protocol/protocol_state/epochs/happy_path_statemachine.go

```diff
@@ -58,12 +60,16 @@ func NewHappyPathStateMachine(view uint64, parentState *flow.RichEpochProtocolSt
 //     CAUTION: the HappyPathStateMachine is left with a potentially dysfunctional state when this error occurs. Do NOT call the Build method
 //     after such error and discard the HappyPathStateMachine!
 func (u *HappyPathStateMachine) ProcessEpochSetup(epochSetup *flow.EpochSetup) (bool, error) {
+	u.consumer.OnServiceEventReceived(epochSetup.ServiceEvent())
 	err := protocol.IsValidExtendingEpochSetup(epochSetup, u.parentState)
 	if err != nil {
```

Comment:
similarly here. I think the error documentation is not strict enough yet to assume that _all_ errors are `InvalidServiceEventError` here (even though I think that is the case for the implementation -- but unless clearly documented, that is a property easily broken by accident)